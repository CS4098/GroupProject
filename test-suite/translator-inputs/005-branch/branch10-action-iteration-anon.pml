process test{
	branch br1 {
		action act1 {
		requires { a }
		provides { e }
		}
		iteration {
            action act2 {
                requires { c }
                provides { d }
            }
            action act3 {
                requires { d }
                provides { b }
            }
        }
	}
}
