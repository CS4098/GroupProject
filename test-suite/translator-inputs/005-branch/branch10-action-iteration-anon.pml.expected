bool a;
bool b;
bool c;
bool d;
bool e;

active proctype test()
{
	int br1 = _nr_pr;
	run act1();
	run anon_proctype_1();
	_nr_pr == br1 ->
}

proctype act1()
{
	a ->
		e = true;
}

proctype anon_proctype_1()
{
    int count__;
    for (count__ : 1 .. 4) {
        c ->
            d = true;
        d ->
            b = true;
    }
}
